pygmt/src/grdcut.py

```diff
@@ -108,4 +113,6 @@ def grdcut(grid, **kwargs):
                     module="grdcut", args=build_arg_string(kwargs, infile=vingrd)
                 )
 
-        return load_dataarray(outgrid) if outgrid == tmpfile.name else None
+        return (
```

Comment:
Changes here will conflict with changes in PR https://github.com/GenericMappingTools/pygmt/pull/2398

Reply:
Yeah, I was thinking whether to wait for that PR to get merged first. We can't use GMT_GRID here though, need to implement GMT_IMAGE for 3-band inputs.

Reply:
PR #2398 has been merged. Will need to resolve conflicts with #3115.

Reply:
With PR #3128 merged, we can continue this PR.

Changes in `pygmt/src/grdcut.py` and `pygmt/io.py` can be reverted or moved into separate PRs. We also no long need to wait for #3115, because the `_load_remote_dataset` function calls the `read` module rather than the `grdcut` module since #3120.

Reply:
Ah I see, thanks for pointing out #3120. I've got some cherrypicked code from #3128 in this branch as I was reviewing that PR locally, and think it was working fine except for the gtype issue you just fixed. I'll revert those `grdcut.py`/`io.py` changes tomorrow morning my time to get a clean PR.

Reply:
Done in 01d1b8014e3f2abf4e349691b286328df65dd377 and 0ca210a52f1951327b167f36c644753c3715a7a6